libs/wingc/src/jsify.rs

```diff
@@ -710,3 +717,11 @@ impl JSifier {
 		format!("{};", self.jsify_symbol(&member.name))
 	}
 }
+
+fn is_mutable_collection(expression: &Expr) -> bool {
+    if let Some(evaluated_type) = expression.evaluated_type.borrow().as_ref() {
+			evaluated_type.is_mutable_collection()
+		} else {
+			false
+		}
+}
```

Comment:
(style) I think it makes sense to associate these kinds of utility function with the target type, but I'll let the king of Rust style @yoav-steinberg weigh in:

```rs
impl Expr {
	fn type(&self) -> TypeRef {
		if let Some(evaluated_type) = expression.evaluated_type.borrow().as_ref() {
			evaluated_type
		} else {
			panic!("expr.type() was called before the expression's type was evaluated");
		}
	}
	fn is_mutable_collection(&self) -> bool {
		this.type().is_mutable_collection()
	}
}
```

Reply:
I'll fast-follow with this after @yoav-steinberg gives his blessing. I rather get this PR merged first if that's okay.